go.mod

```
@@ -1,6 +1,6 @@
 module github.com/nutanix-cloud-native/cluster-api-provider-nutanix
 
-go 1.22.1
+go 1.22.2
```

I think this should stick at 1.21 to be usable by projects at that golang level. If you want to use go 1.22 for builds still you can specify the toolchain to use to build the project specifically. https://go.dev/doc/toolchain is a good resource to understand this. I haven't done this previously myself though but i think this should work...

```suggestion
go 1.21

toolchain go1.22.2
```